backend/substrapp/views/utils.py

```diff
@@ -89,34 +89,62 @@ def download_file(self, request, django_field, ledger_field=None):
             return Response({'message': 'Unauthorized'},
                             status=status.HTTP_403_FORBIDDEN)
 
+        if not ledger_field:
+            ledger_field = django_field
+
         if get_owner() == asset['owner']:
-            obj = self.get_object()
-            data = getattr(obj, django_field)
-            response = CustomFileResponse(
-                open(data.path, 'rb'),
-                as_attachment=True,
-                filename=os.path.basename(data.path)
-            )
+            response = self._download_local_file(django_field)
         else:
-            node_id = asset['owner']
-            auth = authenticate_outgoing_request(node_id)
-            if not ledger_field:
-                ledger_field = django_field
-            r = get_remote_file(asset[ledger_field]['storageAddress'], auth, stream=True)
-            if not r.ok:
-                return Response({
-                    'message': f'Cannot proxify asset from node {asset["owner"]}: {str(r.text)}'
-                }, status=r.status_code)
-
-            response = CustomFileResponse(
-                streaming_content=(chunk for chunk in r.iter_content(512 * 1024)),
-                status=r.status_code)
-
-            for header in r.headers:
-                # We don't use hop_by_hop headers since they are incompatible
-                # with WSGI
-                if not is_hop_by_hop(header):
-                    response[header] = r.headers.get(header)
+            response = self._download_remote_file(ledger_field, asset)
+
+        return response
+
+    def download_local_file(self, request, django_field, ledger_field=None):
```

Comment:
Not related to this PR but `download` is misleading (same for all `download_*` methods). What about `stream`, `forward`, `send`?

Reply:
I agree we could find a better name, for a later refactoring maybe!